### ISSUE TITLE
Address-of (&) treated as ARITH

### ISSUE THREAD

**Issue by Rassi** [uncrustify/uncrustify#1398]
Various uses of address-of operator (&), results in spaces using `sp_arith = add`.

Expected versus output diff:
---
```diff
diff --git "a/C:\\dev\\uncrustify\\tests\\output\\staging\\60037-UNI-29935.cpp" "b/C:\\dev\\uncrustify\\tests\\results\\staging\\60037-UNI-29935.cpp"
index ed615ac..acbbd56 100644
--- "a/C:\\dev\\uncrustify\\tests\\output\\staging\\60037-UNI-29935.cpp"
+++ "b/C:\\dev\\uncrustify\\tests\\results\\staging\\60037-UNI-29935.cpp"
@@ -1,6 +1,6 @@
-void Foo(BarType& x, void BarFunc());
+void Foo(BarType & x, void BarFunc());
 
 void Bar()
 {
-	void BarFunc2(BarType& x);
+	void BarFunc2(BarType & x);
 }
```

Debug support files:
---
[60037-UNI-29935.cpp](https://gist.githubusercontent.com/Rassi/dbe3bd95009056245d20dc9ad33e17c9/raw/4a3bb71f749f18d04cbacd83ebe57d8322c5f7e1/60037-UNI-29935.cpp)
[60037-UNI-29935.cpp.log](https://gist.githubusercontent.com/Rassi/dbe3bd95009056245d20dc9ad33e17c9/raw/3bc3b0b1a1689270c5dba0fb47ab7209c72a3687/60037-UNI-29935.cpp.log)
[60037-UNI-29935.cpp.unc](https://gist.githubusercontent.com/Rassi/dbe3bd95009056245d20dc9ad33e17c9/raw/3ec8ff3243a21689bc4407e08776096d8d1f2511/60037-UNI-29935.cpp.unc)
[UNI-29935.cfg](https://gist.githubusercontent.com/Rassi/dbe3bd95009056245d20dc9ad33e17c9/raw/aeef527d54e85bdca57aee624cbd5ad541825cfa/UNI-29935.cfg)
[UNI-29935.cpp](https://gist.githubusercontent.com/Rassi/dbe3bd95009056245d20dc9ad33e17c9/raw/4a3bb71f749f18d04cbacd83ebe57d8322c5f7e1/UNI-29935.cpp)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by mihaipopescu** [comment]
It also happens with `*` pointers with non-POD types.

**Comment by satishp87** [comment]
In unity fork this issue is addressed with below PR
Unity-Technologies#40

**Comment by guy-maurel** [comment]
Unity-Technologies#40 Doesn't fix the issue.

**Comment by guy-maurel** [comment]
#1616 is a bugfix proposal

**Comment by gmaurel** [comment]
Fixed at eb22c0fbc72562001996ffb215eeff1abc580cd1, on Feb 14 18:07:10 2018
